src/themes/JSONTheme.ts

```diff
@@ -0,0 +1,1003 @@
+/*
+ * Copyright 2012-2020 Online-Go.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// import { createContext } from "react";
+import { GoTheme, GoThemeBackgroundReactStyles, GoThemeBackgroundCSS } from "../GoTheme";
+import { GoThemesInterface } from "../GoThemes";
+import { _ } from "../translate";
+import { GOBAN_FONT } from "../GobanCore";
+//import { deviceCanvasScalingRatio } from "../GoUtil";
+
+/*
+This class makes it easy to design a custom OGS theme using
+json and share it on the interent. It also helps artists
+design image-based themes that need no code, by supplying a JSON url
+
+Eventually (one hopes) users can supply their own JSON themes & share them
+
+*/
+
+/*  A NOTE
+    So, it's complicated.
+    1. A GoTheme is created as one thing (via "class"), but
+    2. GoThemes are then separated by board & stone rendering, and
+    3. GoThemes are never treated like objects, but passed around as classes that components can instantiate
+    4. Every GoTheme (class) must be hardcoded into GoThemes.ts, and is immutable throughout the duration of app run.
```

Comment:
This may be how it's implemented because I didn't have a use case for anything fancier, but if you wanted to dynamically set those that should be just be fine.